packages/jest-runner/src/index.ts

```diff
@@ -164,11 +168,19 @@ class TestRunner {
           },
           globalConfig: this._globalConfig,
           path: test.path,
-        });
+        }) as PromiseWithCustomMessage<TestResult>;
+
+        if (promise.UNSTABLE_onCustomMessage) {
+          promise.UNSTABLE_onCustomMessage(([event, payload]: any) => {
```

Comment:
do we need to type this? should preferably be inferred from `UNSTABLE_onCustomMessage`. _If not_, please avoid `any`